routing/route.hpp

```diff
@@ -135,6 +136,7 @@ class Route
   inline void SetStreetNames(TStreets && v) { m_streets = move(v); }
   inline void SetAltitudes(feature::TAltitudes && v) { m_altitudes = move(v); }
   inline void SetTraffic(vector<traffic::SpeedGroup> && v) { m_traffic = move(v); }
+  inline void SetSegmentInfo(vector<SegmentInfo> && v) {m_segmentInfo = std::move(v);}
```

Comment:
I suggest you to use universal-references here. The code will be almost the same (move must be replaced by forward) but more flexible.